Content.Server/DeviceLinking/Systems/PressurePlateSystem.cs

```diff
@@ -0,0 +1,60 @@
+using Content.Server.DeviceLinking.Components;
+using Content.Shared.DeviceLinking;
+using Content.Shared.Placeable;
+using Content.Server.Physics;
+
+namespace Content.Server.DeviceLinking.Systems;
+
+/// <see cref="PressurePlateComponent"/>
+public sealed class PressurePlateSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly DeviceLinkSystem _signalSystem = default!;
+    [Dependency] private readonly WeightSystem _weight = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PressurePlateComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<PressurePlateComponent, ItemPlacedEvent>(OnItemPlaced);
+        SubscribeLocalEvent<PressurePlateComponent, ItemRemovedEvent>(OnItemRemoved);
+    }
+
+    private void OnItemRemoved(EntityUid uid, PressurePlateComponent component, ref ItemRemovedEvent args)
+    {
+        UpdateState(uid, component);
+    }
+
+    private void OnItemPlaced(EntityUid uid, PressurePlateComponent component, ref ItemPlacedEvent args)
+    {
+        UpdateState(uid, component);
+    }
+
+    private void OnInit(EntityUid uid, PressurePlateComponent component, ComponentInit args)
+    {
+        _signalSystem.EnsureSourcePorts(uid, component.PressedPort, component.ReleasedPort);
+        _appearance.SetData(uid, PressurePlateVisuals.Pressed, false);
+    }
+    private void UpdateState(EntityUid uid, PressurePlateComponent component)
+    {
+        if (!TryComp<ItemPlacerComponent>(uid, out var itemPlacer))
+            return;
+
```

Comment:
Use steptrigger and just subscribe to events.

Reply:
or just add/remove mass in the handlers then update state just checks threshold